IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ElementsHandler.cs

```diff
@@ -91,5 +92,21 @@ public bool ContainsText(string text)
                 return false;
             }
         }
+
+        public IWebElement GetSingleExistingElement(Func<IWebElement, bool> predicate)
+        {
+            IWait<IWebDriver> wait = Wait;
+            wait.IgnoreExceptionTypes(typeof(NoSuchElementException));
+            return wait.Until(d =>
+            {
+                var foundElems = d.FindElements(Locator).Where(predicate).ToList();
+                if(foundElems.Count != 1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(predicate), 
```

Comment:
I am not sure this is really the right exception here since it is not really the argument that is out of range, rather the result of the query that returned too many results. I would match how the LINQ Single method works, and throw an InvalidOperationException in this case.

Reply:
Oh, duh. Thank you for catching that. Fixed and messaging updated!